### PR TITLE
Get data from device for email settings page

### PR DESCRIPTION
**Pull request by alistairjcbrown**
Task/Issue URL: https://app.asana.com/0/72649045549333/1201900580321807/f

### Description
Update to allow requesting user data from Android devices

### Steps to test this PR
1. Update `common/src/main/java/com/duckduckgo/app/global/AppUrl.kt` to have host value align with dev instance
2. Update autofill to https://github.com/duckduckgo/duckduckgo-autofill/pull/52
3. Build browser
4. Go to my dev instance
5. If needed, log in with production duck email username
6. See "Autofill" tab with duck email and private address generator

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/email/EmailManager.kt
Comment (aitorvs) [inline app/src/main/java/com/duckduckgo/app/email/EmailManager.kt:118]: There is not need to create private mutable properties here that also don't go through our dependency injection.
This model is rather simple so I'd rather use `JSONObject`
Also the signature doesn't need to be nullable.

```suggestion
    override fun getUserData(): String {
        return JSONObject().apply { 
            put("token", emailDataStore.emailToken)
            put("userName", emailDataStore.emailUsername)
            put("nextAlias", emailDataStore.nextAlias?.replace(DUCK_EMAIL_DOMAIN, ""))
        }.toString()
    }

```

File: app/src/main/java/com/duckduckgo/app/email/EmailManager.kt
Comment (aitorvs) [inline app/src/main/java/com/duckduckgo/app/email/EmailManager.kt:46]: nit: Signature doesn't need to be nullable
```suggestion
    fun getUserData(): String
```

File: app/src/main/java/com/duckduckgo/app/email/EmailJavascriptInterface.kt
Comment (aitorvs) [inline app/src/main/java/com/duckduckgo/app/email/EmailJavascriptInterface.kt:50]: Signature doesn't need to be nullable
```suggestion
    fun getUserData(): String {
```

**Review by aitorvs — CHANGES_REQUESTED** [review]
I left some comments. Most important is about creating unnecessary public properties without using dependency injection and unnecessary data models like `UserData`

Also, can we remove the last commit that modifies the version of autofill? Bumping autofill should be in its own PR.

**Comment by alistairjcbrown** [comment]
Thanks so much for reviewing @aitorvs 🙇
I've updated for your comments, and dropped the last commit -- how does this look now?

**Review by aitorvs — COMMENTED** [review]
This LGTM code wise. Haven't tested it end to end. @marcosholgado can you help with that?

**Review by aitorvs — APPROVED** [review]

**Comment by marcosholgado** [comment]
Adding a do not merge label, I will add unit tests before merging.

**Comment by malmstein** [comment]
https://github.com/duckduckgo/Android/actions/runs/1909666883 has been running for 3 hours

**Review by marcosholgado — APPROVED** [review]
Already approved but LGTM. I added unit tests for the new code and a minor change to add the json constants to the companion object. Thanks @alistairjcbrown!

**Pull request merged**
